ddtrace/tracer/spancontext_test.go

```diff
@@ -245,6 +246,196 @@ func TestSpanFinishPriority(t *testing.T) {
+			wantPeerServiceSource:       "db.instance",
+			wantPeerServiceRemappedFrom: "",
+		},
+		{
+			name: "DBClientDefaultsDisabled",
+			spanOpts: []StartSpanOption{
+				Tag("span.kind", "client"),
+				Tag("db.system", "some-db"),
+				Tag("db.instance", "db-instance"),
+			},
+			peerServiceDefaultsEnabled:  false,
+			peerServiceMappings:         nil,
+			wantPeerService:             "",
+			wantPeerServiceSource:       "",
+			wantPeerServiceRemappedFrom: "",
+		},
+		{
+			name: "DBCassandra",
+			spanOpts: []StartSpanOption{
+				Tag("span.kind", "client"),
+				Tag("db.system", "cassandra"),
+				Tag("db.instance", "db-instance"),
+				Tag("db.cassandra.contact.points", "h1,h2,h3"),
+				Tag("out.host", "out-host"),
+			},
+			peerServiceDefaultsEnabled:  true,
+			peerServiceMappings:         nil,
+			wantPeerService:             "h1,h2,h3",
+			wantPeerServiceSource:       "db.cassandra.contact.points",
+			wantPeerServiceRemappedFrom: "",
+		},
+		{
+			name: "GRPCClient",
+			spanOpts: []StartSpanOption{
+				Tag("span.kind", "client"),
+				Tag("rpc.system", "grpc"),
+				Tag("rpc.service", "rpc-service"),
+				Tag("out.host", "out-host"),
+			},
+			peerServiceDefaultsEnabled:  true,
+			peerServiceMappings:         nil,
+			wantPeerService:             "rpc-service",
+			wantPeerServiceSource:       "rpc.service",
+			wantPeerServiceRemappedFrom: "",
+		},
+		{
+			name: "OtherClients",
+			spanOpts: []StartSpanOption{
+				Tag("span.kind", "client"),
+				Tag("out.host", "out-host"),
+			},
+			peerServiceDefaultsEnabled:  true,
+			peerServiceMappings:         nil,
+			wantPeerService:             "out-host",
+			wantPeerServiceSource:       "out.host",
+			wantPeerServiceRemappedFrom: "",
+		},
+		{
+			name: "WithMapping",
+			spanOpts: []StartSpanOption{
+				Tag("span.kind", "client"),
+				Tag("out.host", "out-host"),
+			},
+			peerServiceDefaultsEnabled: true,
+			peerServiceMappings: map[string]string{
+				"out-host": "remapped-out-host",
+			},
+			wantPeerService:             "remapped-out-host",
+			wantPeerServiceSource:       "out.host",
+			wantPeerServiceRemappedFrom: "out-host",
+		},
+		{
+			// in this case we skip defaults calculation but track the source and run the remapping.
+			name: "WithoutSpanKindAndPeerService",
+			spanOpts: []StartSpanOption{
+				Tag("peer.service", "peer-service"),
+			},
+			peerServiceDefaultsEnabled: false,
+			peerServiceMappings: map[string]string{
+				"peer-service": "remapped-peer-service",
+			},
+			wantPeerService:             "remapped-peer-service",
+			wantPeerServiceSource:       "peer.service",
+			wantPeerServiceRemappedFrom: "peer-service",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tracer.config.peerServiceDefaultsEnabled = tc.peerServiceDefaultsEnabled
+			tracer.config.peerServiceMappings = tc.peerServiceMappings
+			s := tracer.StartSpan("peer-service-test", tc.spanOpts...)
+			s.Finish()
```

Comment:
Update the test to make this the child of another span. You'll find that most of these tests fail.

Reply:
updated and fixed, thanks for catching this! :)